controllers/hetznercluster_controller.go

```diff
@@ -344,9 +344,15 @@ func (r *HetznerClusterReconciler) reconcileDelete(ctx context.Context, clusterS
 		return reconcile.Result{}, fmt.Errorf("failed to delete load balancers for HetznerCluster %s/%s: %w", hetznerCluster.Namespace, hetznerCluster.Name, err)
 	}
 
-	// delete the network
-	if err := network.NewService(clusterScope).Delete(ctx); err != nil {
-		return reconcile.Result{}, fmt.Errorf("failed to delete network for HetznerCluster %s/%s: %w", hetznerCluster.Namespace, hetznerCluster.Name, err)
+	// delete the network only if it is owned by us
```

Comment:
this logic should be integrated into the Delete function that we call from the "network" package. Here we should just do a simple function call